app/controllers/GenericExplorerController.scala

```diff
@@ -95,6 +95,54 @@ class GenericExplorerController @Inject() (
     future.map( i => Ok( Json.toJson( i ) ) )
   }
 
+  //Search for nodes with a property in a graph
+  def retrieveNodeProperty( property: String ) = ProfileFilterAction( jwtVerifier.get ).async { implicit request =>
+    Logger.debug( "Request to retrieve node(s) with property " + property )
+    val g = graphTraversalSource
+    val t = g.V().has( property )
+
+    val future: Future[List[PersistedVertex]] = {
+      if ( t.hasNext ) {
+        Future.sequence(
+          for ( vertex <- t.asScala.toList ) yield vertexReader.read( vertex )
+        )
+      }
+
+      else
+        Future.successful( List() )
+    }
+    future.map( i => Ok( Json.toJson( i ) ) )
+  }
+
+  def getValuesForProperty( property: String ) = ProfileFilterAction( jwtVerifier.get ).async { implicit request =>
+    Logger.debug( "Request to retrieve values for property " + property )
+
+    val g = graphTraversalSource
+    val t = g.V().values[String]( property ).asScala.toList
```

Comment:
Have you tried with `system:creation_time`?
I think it won't go past the type cast  here.

Reply:
The thing here is to make it generic, so that it can be used with every type.  I suppose we can set the value to Any.

Reply:
Except that will create issues wiht the Json serialisation.